src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1285,6 +1285,85 @@ internal bool HasCollectionExpressionApplicableAddMethod(SyntaxNode syntax, Type
             }
         }
 
+        internal bool HasCollectionExpressionAddMethod(SyntaxNode syntax, TypeSymbol targetType)
+        {
+            const string methodName = "Add";
+            var useSiteInfo = CompoundUseSiteInfo<AssemblySymbol>.Discarded;
+
+            var lookupResult = LookupResult.GetInstance();
+            LookupInstanceMember(lookupResult, targetType, leftIsBaseReference: false, rightName: methodName, rightArity: 0, invoked: true, ref useSiteInfo);
```

Comment:
>LookupInstanceMember [](http://example.com/codeflow?start=12&length=20)

Rather that trying to duplicate the lookup code, I think we should use ```Binder.BindInstanceMemberAccess``` and proceed only if we get a ```BoundMethodGroup``` back. I am thinking of a call similar to the one in ```HasCollectionExpressionApplicableAddMethod.makeInvocationExpression```. Afterwards, I think we should pay attention to the ```BoundMethodGroup.SearchExtensionMethods``` value before looking for extensions.